src/wallet/db.cpp

```diff
@@ -697,7 +697,6 @@ void BerkeleyEnvironment::Flush(bool fShutdown)
                 if (!fMockDb) {
                     fs::remove_all(fs::path(strPath) / "database");
                 }
-                g_dbenvs.erase(strPath);
```

Comment:
In commit "Move BerkeleyEnvironment deletion from internal method to callsite" (a769461d5e37ddcb771ae836254fdc69177a28c4)

@achow101 do remember if this change was this done for specific reason? It seems like an odd thing to include here.

Reply:
IIRC there are two reasons for this. The first was that we needed this in the next commit to avoid having a destroyed db environment because we want to reopen it. The second reason that I don't think it is good for an object to destroy its own references as that's what it does here.